docs/advanced-features/i18n-routing.md

```diff
@@ -0,0 +1,134 @@
+# Internationalized Routing
+
+<details>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/i18n-routing">i18n routing</a></li>
+  </ul>
+</details>
+
+Next.js has built in support for i18n routing since v9.5.7. With the built-in support you can provide a list of locales you want to support, the default locale, and domain specific locales and Next.js will help handle the routing for these values.
+
+The i18n routing support is currently meant to compliment existing i18n library solutions like react-intl, react-i18next, lingui, rosetta, and others by streamlining the routes and locale parsing aspects of i18n.
+
+## How to get started
+
+To get started you need to add the `i18n` config to your `next.config.js` file.
+
+```js
+module.exports = {
+  i18n: {
+    // These are all the locales you want to be able to support in
+    // your application
+    locales: ['en-US', 'fr', 'nl-NL'],
+    // This is the default locale you want to be used when visiting
+    // a non-local prefixed path e.g. `/hello`
+    defaultLocale: 'en-US',
+    // These are a list of locale domains and the default locale they
+    // should handle (these are not required)
+    domains: [
+      {
+        domain: 'example.com',
+        defaultLocale: 'en-US',
+      },
+      {
+        domain: 'example.nl',
+        defaultLocale: 'nl-NL',
+      },
+      {
+        domain: 'example.fr',
+        defaultLocale: 'fr',
+      },
+    ],
+  },
+}
+```
+
+## How are the routes handled
+
+Using the above config when a user visits `/` Next.js will automatically try to detect which locale the user prefers with the [`Accept-Language`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) header and the current domain.
+
+If a locale other than the default locale is detected the user will be redirected to either:
+
+a) the domain with that locale specified as the default
+b) the locale prefixed path if no domain matched
+
+For example, if a user with the `Accept-Language` header `fr;q=0.9` visits `example.com` they will be redirected to `example.fr` since that domain handles the `fr` locale by default.
```

Comment:
```suggestion
For example, if a user with the `Accept-Language` header `fr;q=0.9` visits `example.com`, they will be redirected to `example.fr` since that domain handles the `fr` locale by default.
```